src/cluster_legacy.c

```diff
@@ -1252,6 +1252,57 @@ void clusterInitLast(void) {
 
 /* Called when a cluster node receives SHUTDOWN. */
 void clusterHandleServerShutdown(void) {
+    if (nodeIsPrimary(myself) && server.auto_failover_on_shutdown) {
```

Comment:
I would suggest moving this non-trivial block of code into its own function for readability.